torch/_dynamo/variables/base.py

```diff
@@ -185,9 +185,53 @@ def const_getattr(self, tx, name: str) -> Any:
         """getattr(self, name) returning a python constant"""
         raise NotImplementedError()
 
+    def dynamic_getattr(self, tx, name):
+        if not self.source:
+            raise NotImplementedError()
+
+        if not is_from_local_source(self.source):
+            raise NotImplementedError()
+
+        # For local source, we associate the real value. We use this real value
+        # for implementing getattr fallthrough on the variable tracker base class.
+
+        # Note - this scope construction is mirrored in guards
+        # A subsequent PR will introduce a util.
+        scope = {"L": tx.output.local_scope, "G": tx.output.global_scope}
+        try:
+            _input_associated_real_value = eval(self.source.name(), scope)
+        except Exception:
+            # TODO(voz): Check in on this and find owner of SuperSource.
```

Comment:
I added SuperSource a while back
https://github.com/pytorch/pytorch/commit/8e2e648f846f21aae7a8eeadf6964d8a9467af1b
The motivation chain was 
 - supporting default arguments required valid source chains for the function that has the default arg
 - some functions were created off of SuperVariables
 - we need a valid source for a SuperVariable
 
 do you have more info about what's broken?

Reply:
I am not actually sold on super source always producing valid code! There are cases where it made code that failed to eval (The requirement for sources in dynamo)

Reply:
I am working a PR to add this invariant atm.